### PR TITLE
[test/spec] Spec tests for nocasematch

### PR DESCRIPTION
**Pull request by Crestwave**
I wasn't entirely sure what to put, but I figured it would be better to be too thorough than not thorough enough.

### CODE REVIEW

File: test/spec.sh [hunk 1]
```diff
@@ -709,6 +709,11 @@ extglob-match() {
     $BASH $MKSH $OSH_LIST "$@"
 }
 
+nocasematch-match() {
+  sh-spec spec/nocasematch-match.test.sh \
```
Comment (andychu) [inline test/spec.sh:713]: Please `add --osh-failures-allowed 6` here.  That just indicates "we know about these failures, the whole test suite shouldn't fail"

Comment (Crestwave) [reply test/spec.sh:713]: Just to clarify, it only affects `case` and `[[` and the rest are to verify that it does not affect it.  As such, OSH currently passes 3 of the tests, so should it be `--osh-feailures-allowed 3`?

Comment (andychu) [reply test/spec.sh:713]: Ah yes, it should be 3.  You can leave the other tests since the look informative.

**Comment by andychu** [comment]
Great thanks!

**Pull request merged**